src/main/java/org/prebid/server/bidder/ix/IxBidder.java

```diff
@@ -1,30 +1,190 @@
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.Banner;
 import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Format;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.request.Publisher;
+import com.iab.openrtb.request.Site;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.Json;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.BidderUtil;
 import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
 import org.prebid.server.bidder.model.HttpCall;
 import org.prebid.server.bidder.model.HttpRequest;
 import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ix.ExtImpIx;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
 
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
  * ix {@link Bidder} implementation.
  */
-public class IxBidder implements Bidder {
+public class IxBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpIx>> IX_EXT_TYPE_REFERENCE = new
+            TypeReference<ExtPrebid<?, ExtImpIx>>() {
+            };
+
+    // maximum number of bid requests
+    private static final int REQUEST_LIMIT = 20;
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+
+    private final String endpointUrl;
+
+    public IxBidder(String endpointUrl) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+    }
 
     @Override
-    public Result<List<HttpRequest>> makeHttpRequests(BidRequest bidRequest) {
-        throw new UnsupportedOperationException();
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        if (bidRequest.getApp() != null) {
+            return Result.emptyWithError(BidderError.badInput("ix doesn't support apps"));
+        }
+
+        final List<BidderError> errors = new ArrayList<>();
+        final List<BidRequest> prioritizedRequests = new ArrayList<>();
+        final List<BidRequest> regularRequests = new ArrayList<>();
+        for (Imp imp : bidRequest.getImp()) {
+            if (prioritizedRequests.size() == REQUEST_LIMIT) {
+                break;
+            }
+            try {
+                validateImp(imp);
+                final ExtImpIx extImpIx = parseAndValidateIxExt(imp);
```

Comment:
i think we can:
- move calls `parseAndValidateIxExt(...)` and `modifyRequest()` into `makeRequests` method.
- then there we can create request builder and pass it to `modifyRequest`.
- also i recommend to rename `modifyRequest` to `modifySite`